Added AWS security documentation.

Laren-AWS opened the pull request:
This documentation describes AWS security and division of security responsibilities, and provides guidance and links for specific security topics.

jamesls commented on docs/source/guide/_security-includes.txt:
```diff
@@ -0,0 +1,15 @@
+.. |SERVICENAME| replace:: Boto 3
```
Some of these replacements don't make sense or seem unnecessary.  For example, `Boto 3` isn't really a service name, and some like `|AWS| replace:: AWS` and `|IAM| replace:: IAM` don't currently do anything.  Additionally, some such as `|MCELong|` seem opaque compared to `Amazon Macie` and is only referenced a single time.

Personally, I'd prefer we don't use `replace` at all as I think it makes it harder to read the raw source of `security.rst` for not a whole lot of benefit, and I'd prefer to optimize the readability of the source.

But if you'd prefer to use them, I'd say keep them directly in `security.rst` and don't use an `include`.  We can always factor this out to a separate file if there are multiple docs that start to use these `replace` values.

Laren-AWS commented:
I agree on the replacements. I removed them all and updated the content to use the actual strings instead. Let me know if you need anything else, thanks!
